core/src/Kokkos_BitManipulation.hpp

```diff
@@ -278,6 +304,34 @@ KOKKOS_IMPL_HOST_FUNCTION
 
 #undef KOKKOS_IMPL_USE_GCC_BUILT_IN_FUNCTIONS
 
+template <class T>
+KOKKOS_FUNCTION T rotl_builtin_host(T x, int s) noexcept {
+  return rotl(x, s);
+}
+
+template <class T>
+KOKKOS_FUNCTION T rotl_builtin_device(T x, int s) noexcept {
+#ifdef KOKKOS_ENABLE_SYCL
+  return sycl::rotate(x, s);
```

Comment:
Did not compile because the two arguments have different types
https://github.com/intel/llvm/blob/2d47764828724b5c4e11384eb110bde427dcbd25/sycl/include/sycl/builtins.hpp#L1019-L1024

Talked with Daniel and will just use the fallback instead.